trin-core/src/portalnet/discovery.rs

```diff
@@ -53,20 +53,24 @@ impl Discovery {
     pub fn new(portal_config: PortalnetConfig) -> Result<Self, String> {
         let listen_all_ips = SocketAddr::new("0.0.0.0".parse().unwrap(), portal_config.listen_port);
 
-        let ip_addr = if portal_config.internal_ip {
-            socket::default_local_address(portal_config.listen_port)
+        let (ip_addr, ip_port) = if portal_config.internal_ip {
+            (None, portal_config.listen_port)
         } else {
-            portal_config
+            let known_external = portal_config
```

Comment:
in `cli.rs` restrict arguments to only `--internal-ip` or `--external-address ...`